Mage/src/main/java/mage/game/combat/CombatGroup.java

```diff
@@ -720,4 +791,29 @@ public boolean changeDefenderPostDeclaration(UUID newDefenderId, Game game) {
         }
         return false;
     }
+
+    public boolean defenderControlsDefensiveFormation(Game game) {
+        // for handling Defensive Formation
+        for (Permanent defensiveFormation : game.getBattlefield().getAllActivePermanents(defendingPlayerId)) {
+            if (defensiveFormation.getAbilities().containsKey(ControllerAssignCombatDamageToBlockersAbility.getInstance().getId())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public boolean isButcherOrgg(Permanent creature, UUID playerId, boolean first, Game game, boolean isAttacking) {
+        // for handling Butcher Orgg
+        if (creature.getAbilities().containsKey(ControllerDivideCombatDamageAbility.getInstance().getId())) {
+            Player player = game.getPlayer(defenderControlsDefensiveFormation(game) ? defendingPlayerId : playerId);
+            // 10/4/2004 	If it is blocked but then all of its blockers are removed before combat damage is assigned, then it won’t be able to deal combat damage and you won’t be able to use its ability.
+            if (!((blocked && blockers.isEmpty() && isAttacking) || (attackers.isEmpty() && !isAttacking))) {
+                if (player.chooseUse(Outcome.Damage, "Do you wish to divide " + creature.getLogName() + "'s combat damage among defending player and/or any number of defending creatures?", null, game)) {
+                    butcherOrggDamage(creature, player, first, game, isAttacking);
```

Comment:
Very bad code. Your function called "isButcherOrgg" (e.g. checking function), but you do damage in that "check". Do not use that. Check and assign damage in separate functions.